### PR TITLE
refactor(components): refactor button

### PR DESCRIPTION
**Pull request by sxzz**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/button/src/button.vue [hunk 1]
```diff
@@ -48,120 +48,89 @@ import {
...
+const emit = defineEmits(buttonEmits)
+const slots = useSlots()
+
+const buttonGroupContext = inject(buttonGroupContextKey, undefined)
+const globalConfig = useGlobalConfig('button')
+const ns = useNamespace('button')
+const { form } = useFormItem()
+const _size = useSize(computed(() => buttonGroupContext?.size))
+const _disabled = useDisabled()
+const _ref = ref()
```
Comment (JeremyWuuuuu) [inline packages/components/button/src/button.vue:67]: Can we add annotation for this one to refer to `HTMLButtonElement`? Which enriches the typing system

File: packages/hooks/__tests__/use-form-item.spec.ts [hunk 2]
```diff
@@ -36,15 +36,15 @@ const mountComponent = (setup = NOOP, options = {}) => {
 
 const getButtonVm = (wrapper: ReturnType<typeof mountComponent>) => {
   return wrapper.findComponent(ElButton).vm as any as {
```
Comment (JeremyWuuuuu) [inline packages/hooks/__tests__/use-form-item.spec.ts:38]: **Question:** why do we need to cast the type here? Shouldn't `ElButton` be type annotated already?

Comment (sxzz) [reply packages/hooks/__tests__/use-form-item.spec.ts:38]: Exported variables are not attached to the Button type.

File: packages/components/button/src/button.vue (hunk 1 above)
Comment (sxzz) [reply packages/components/button/src/button.vue]: Fixed.

**Review by zhangsanfe — APPROVED** [review]

File: packages/hooks/__tests__/use-form-item.spec.ts (hunk 2 above)
Comment (JeremyWuuuuu) [reply packages/hooks/__tests__/use-form-item.spec.ts:38]: Just as you mentioned it is not attached, should we try test the variable via other options like in the code, `_size` is used here: `n.m(_size)`, shouldn't we test this by finding whether the variable exists or not? For instance:

```ts
expect(wrapper.find('el-button--default').exists()).toBeTrue()
```

**Review by jw-foss — APPROVED** [review]

**Comment by JeremyWuuuuu** [comment]
LGTM! Thanks!

**Pull request merged**